### ISSUE TITLE
TLS negotiation failure — ldap_sasl_bind(SIMPLE): Can't contact LDAP server (-1)

### ISSUE THREAD

**Issue by fooness** [bitnami/bitnami-docker-openldap#110]
### Name and Version

`bitnami/openldap:latest` — $OpenLDAP: slapd 2.6.2 (May  5 2022 12:08:10)

### What steps will reproduce the bug?

We are trying to successfully enable TLS for OpenLDAP to be able to use the secure `ldaps://` protocol for all our LDAP communications/connections. The to-be-used SSL certificates, issued via `certbot`/`letsencrypt` by `traefik`, and dumped via `ldez/traefik-certs-dumper`, are most likely valid and not the problem here.

For `LDAP_TLS_CA_FILE` we tried using the `bitnami/openldap` internal `/etc/ssl/certs/ca-certificates.crt` … maybe this is — at least, part of — the problem here. We also tried using the host operating system’s `/etc/ssl/certs/ca-certificates.crt` without success; this resulted in the same errors/logs.

Unsecured LDAP connections via the `ldap://` protocol and port `1389` work flawlessly.

Secured LDAP connections via the `ldaps://` protocol and port `1636` result in the errors/logs below.

### What is the expected behavior?

Secured LDAP connections via the `ldaps://` protocol and port `1636` should be working flawlessly, too.

### What do you see instead?

```
$ ldapsearch -x -H ldaps://openldap.example.com:1636 -b "dc=example,dc=com" -D "cn=admin,dc=example,dc=com" -W
Enter LDAP Password:
ldap_sasl_bind(SIMPLE): Can't contact LDAP server (-1)
```

```
$ docker logs -f 8ceffc9b6305
 16:21:09.25 INFO  ==> ** Starting LDAP setup **
 16:21:09.27 INFO  ==> Validating settings in LDAP_* env vars
 16:21:09.28 INFO  ==> Initializing OpenLDAP...
 16:21:09.28 DEBUG ==> Ensuring expected directories/files exist...
 16:21:09.29 INFO  ==> Using persisted data
 16:21:09.29 INFO  ==> ** LDAP setup finished! **

 16:21:09.31 INFO  ==> ** Starting slapd **
62a0cc75.12f1a7b4 0x7f88c1365740 @(#) $OpenLDAP: slapd 2.6.2 (May  5 2022 12:08:10) $
        @f200d7a44f23:/bitnami/blacksmith-sandox/openldap-2.6.2/servers/slapd
62a0cc75.13678bed 0x7f88c1365740 slapd starting
62a0cc82.3320aa1c 0x7f88bfbc3700 conn=1000 fd=14 ACCEPT from IP=XXX.XXX.XXX.XXX:XXXXX (IP=0.0.0.0:1636)
```

### Additional information

```
$ docker ps
CONTAINER ID   IMAGE                                           COMMAND                  CREATED        STATUS                  PORTS                                                                                  NAMES
8ceffc9b6305   bitnami/openldap:latest                         "/opt/bitnami/script…"   16 hours ago   Up 16 hours             0.0.0.0:1389->1389/tcp, :::1389->1389/tcp, 0.0.0.0:1636->1636/tcp, :::1636->1636/tcp   openldap
2807a48b3689   traefik:latest                                  "/entrypoint.sh --ap…"   16 hours ago   Up 16 hours             0.0.0.0:80->80/tcp, :::80->80/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp               traefik
```

```
$ cat docker-compose.yml
version: "3"

services:

  traefik:
    image: traefik:latest
    container_name: traefik
    restart: unless-stopped
    command:
      - "--api.dashboard=true"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.web.http.redirections.entrypoint.to=websecure"
      - "--entrypoints.web.http.redirections.entrypoint.scheme=https"
      - "--entrypoints.websecure.address=:443"
      - "--entrypoints.ldap-tcp.address=:1389/tcp"
      - "--entrypoints.ldap-udp.address=:1389/udp"
      - "--entrypoints.ldaps.address=:1636/tcp"
      - "--certificatesresolvers.myresolver.acme.tlschallenge=true"
      #- "--certificatesresolvers.myresolver.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory"
      - "--certificatesresolvers.myresolver.acme.email=letsencrypt@example.com"
      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
    labels:
      traefik.enable: "true"
      traefik.http.routers.api.entrypoints: "websecure"
      traefik.http.routers.api.tls.certresolver: "myresolver"
      traefik.http.routers.api.rule: "Host(`traefik.example.com`)"
      traefik.http.routers.api.service: "api@internal"
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./data/letsencrypt:/letsencrypt
      - /var/run/docker.sock:/var/run/docker.sock:ro

  traefik-certs-dumper:
    image: ldez/traefik-certs-dumper:latest
    container_name: traefik-certs-dumper
    restart: unless-stopped
    entrypoint: /bin/sh -c 'apk add jq; while ! [ -e /data/acme.json ] || ! [ `jq ".[] | .Certificates | length" /data/acme.json` != 0 ]; do sleep 1; done && traefik-certs-dumper file --version v2 --source /data/acme.json --watch --dest /dump --domain-subdir'
    volumes:
      - ./data/letsencrypt:/data:ro
      - ./data/traefik-certs-dumper:/dump

  openldap--fix-permissions:
    image: bitnami/openldap:latest
    container_name: openldap--fix-permissions
    restart: "no"
    command: "chown -R 1001:1001 /bitnami"
    user: "root"
    volumes:
      - ./data/openldap:/bitnami

  openldap:
    image: bitnami/openldap:latest
    container_name: openldap
    restart: unless-stopped
    depends_on:
      - traefik
      - postgres
    environment:
      BITNAMI_DEBUG: "true"
      LDAP_LOG_LEVEL: "9"
      LDAP_PORT_NUMBER: "1389"
      LDAP_ROOT: "dc=example,dc=org"
      LDAP_ADMIN_USERNAME: "admin"
      LDAP_ADMIN_PASSWORD: "adminpassword"
      LDAP_USERS: "user01,user02"
      LDAP_PASSWORDS: "bitnami1,bitnami2"
      LDAP_USER_DC: "users"
      LDAP_GROUP: "readers"
      LDAP_ENABLE_TLS: "yes"
      LDAP_LDAPS_PORT_NUMBER: "1636"
      LDAP_TLS_CERT_FILE: "/etc/letsencrypt/live/openldap.example.com/certificate.crt"
      LDAP_TLS_KEY_FILE: "/etc/letsencrypt/live/openldap.example.com/privatekey.key"
      LDAP_TLS_CA_FILE: "/etc/ssl/certs/ca-certificates.crt"
      #LDAP_TLS_CA_FILE: "/bitnami/ca-certificates.crt"
      #LDAP_TLS_DH_PARAMS_FILE: ""
    labels:
      traefik.enable: "true"
      traefik.tcp.routers.ldap-tcp.rule: "HostSNI(`*`)"
      traefik.tcp.routers.ldap-tcp.entrypoints: "ldap-tcp"
      traefik.udp.routers.ldap-udp.entrypoints: "ldap-udp"
      traefik.tcp.routers.ldaps.rule: "HostSNI(`*`)"
      traefik.tcp.routers.ldaps.entrypoints: "ldaps"
      traefik.tcp.routers.ldaps.tls: "true"
      traefik.tcp.routers.ldaps.tls.passthrough: "true"
      traefik.tcp.routers.ldaps.tls.certresolver: "myresolver"
    ports:
      - "1389:1389"
      - "1636:1636"
    volumes:
      - ./data/openldap:/bitnami/openldap
      - ./data/traefik-certs-dumper:/etc/letsencrypt/live:ro
      #- type: bind
      #  source: /etc/ssl/certs/ca-certificates.crt
      #  target: /bitnami/ca-certificates.crt

networks:
  default:
    driver: bridge
```

The following post which was found on the traefik community forum was ~quite~ very helpful for getting `bitnami/openldap` working together with `traefik` … https://community.traefik.io/t/issues-with-tls-enabled-tcp-router-to-openldap/3949/13

**Comment by dgomezleon** [comment]
Hello @fooness ,

Could you confirm if it works for you with a simpler configuration first? Something like:

```
version: '2'

services:
  openldap:
    image: docker.io/bitnami/openldap:2.6
    ports:
      - '1389:1389'
      - '1636:1636'
    environment:
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_USERS=user01,user02
      - LDAP_PASSWORDS=password1,password2
      - BITNAMI_DEBUG=true
      - LDAP_LOG_LEVEL=9
      - LDAP_PORT_NUMBER=1389
      - LDAP_ROOT=dc=example,dc=org
      - LDAP_USER_DC="users"
      - LDAP_GROUP="readers"
      - LDAP_ENABLE_TLS=yes
      - LDAP_LDAPS_PORT_NUMBER=1636
      - LDAP_TLS_CERT_FILE: "/etc/letsencrypt/live/openldap.example.com/certificate.crt"
      - LDAP_TLS_KEY_FILE: "/etc/letsencrypt/live/openldap.example.com/privatekey.key"
      - LDAP_TLS_CA_FILE: "/etc/ssl/certs/ca-certificates.crt"
      #LDAP_TLS_CA_FILE: "/bitnami/ca-certificates.crt"
    volumes:
     - ./data/openldap:/bitnami/openldap
      - ./data/traefik-certs-dumper:/etc/letsencrypt/live:ro

volumes:
  openldap_data:
    driver: local
```

**Comment by fooness** [comment]
Hello @dgomezleon, thank you for your reply.

> Could you confirm if it works for you with a simpler configuration first? Something like: […]

I completely shut down our `docker-compose` stack and started up the one you provided; the three lines matching `LDAP_TLS_*_FILE: "…"` had to be converted to `LDAP_TLS_*_FILE=…` to get it up and running.

The result is the same as with our `docker-compose` stack; `ldap://` on port `1389` works, `ldaps://` on port `1636` results in a TLS negotiation failure.

I think the problem might be the ca-certificate, and/or how it “works together” with the dumped `letsencrypt` certificates. Unfortunately, for the time being, I couldn’t figure out an easy way on how to add another service to our (or your) `docker-compose.yml` which itself creates new certificates via e.g. `letsencrypt` AND also somehow creates/downloads/links/whatever a valid (whatever that means exactly) certificate authority file.

I don’t _think_ the problem lies solely in the dumped `letsencrypt` certificates; _maybe_ the problem lies solely in the `ca-certificates.crt` file from within the `bitnami/openldap` container.

How would one create/download/link a definitely valid certificate authority file?

**Comment by dgomezleon** [comment]
Hi @fooness ,


> I completely shut down our docker-compose stack and started up the one you provided; the three lines matching LDAP_TLS_*_FILE: "…" had to be converted to LDAP_TLS_*_FILE=… to get it up and running.

Sorry, it was a copy/paste typo after replacing my values with yours.


This worked for me as a local test:

```
version: '2'

services:
  openldap:
    image: docker.io/bitnami/openldap:2.6
    ports:
      - '1389:1389'
      - '1636:1636'
    environment:
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_USERS=user01,user02
      - LDAP_PASSWORDS=password1,password2
      - BITNAMI_DEBUG=true
      - LDAP_LOG_LEVEL=9
      - LDAP_PORT_NUMBER=1389
      - LDAP_ROOT=dc=example,dc=org

      - LDAP_USER_DC="users"
      - LDAP_GROUP="readers"
      - LDAP_ENABLE_TLS=yes
      - LDAP_LDAPS_PORT_NUMBER=1636
      - LDAP_TLS_CERT_FILE=/opt/bitnami/openldap/certs/DOMAIN.crt
      - LDAP_TLS_KEY_FILE=/opt/bitnami/openldap/certs/DOMAIN.key
      - LDAP_TLS_CA_FILE=/opt/bitnami/openldap/certs/DOMAIN.issuer.crt
      #LDAP_TLS_CA_FILE: "/bitnami/ca-certificates.crt"
      #LDAP_TLS_DH_PARAMS_FILE: ""
    volumes:
      - 'openldap_data:/bitnami/openldap'
      - '/Users/dgomezleon/lets-encrypt/:/opt/bitnami/openldap/certs'

volumes:
  openldap_data:
    driver: local
```

Inside the container, this worked fine:

```
 echo "127.0.0.1 david-wp.dev.closedstack.org" >> /etc/hosts
ldapsearch -x -H ldaps://david-wp.dev.closedstack.org:1636 -b "dc=example,dc=org" -D "cn=admin,dc=example,dc=org" -W
```

As you can see, I used the issuer file for CA authority. My understanding is that you need to use letsencrypt CA file for `LDAP_TLS_CA_FILE` since it is your certificate provider. You can find root and intermediate certificates [here](https://letsencrypt.org/certificates/), or use the issuer file if you generated it.

I hope it helps

**Comment by fooness** [comment]
Thank you for your efforts and testing.

> As you can see, I used the issuer file for CA authority. My understanding is that you need to use letsencrypt CA file for `LDAP_TLS_CA_FILE` since it is your certificate provider. You can find root and intermediate certificates [here](https://letsencrypt.org/certificates/), or use the issuer file if you generated it.

I tried using three different certificates from the linked Let’s Encrypt website: `lets-encrypt-r3.pem`, `isrg-root-x1-cross-signed.pem`, `isrgrootx1.pem` — unfortunately, no success with these, too.

```
version: '2'

services:
  openldap:
    image: docker.io/bitnami/openldap:2.6
    ports:
      - '1389:1389'
      - '1636:1636'
    environment:
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_USERS=user01,user02
      - LDAP_PASSWORDS=password1,password2
      - BITNAMI_DEBUG=true
      - LDAP_LOG_LEVEL=9
      - LDAP_PORT_NUMBER=1389
      - LDAP_ROOT=dc=example,dc=org
      - LDAP_USER_DC="users"
      - LDAP_GROUP="readers"
      - LDAP_ENABLE_TLS=yes
      - LDAP_LDAPS_PORT_NUMBER=1636
      - LDAP_TLS_CERT_FILE=/opt/bitnami/openldap/etc/certs/certificate.crt
      - LDAP_TLS_KEY_FILE=/opt/bitnami/openldap/etc/certs/privatekey.key
      #- LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/lets-encrypt-r3.pem
      #- LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/isrg-root-x1-cross-signed.pem
      - LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/isrgrootx1.pem
    volumes:
      - ./data/openldap:/bitnami/openldap
      - ./data/letsencrypt:/opt/bitnami/openldap/etc/letsencrypt:ro
      - ./data/traefik-certs-dumper/openldap.example.com:/opt/bitnami/openldap/etc/certs:ro

#volumes:
#  openldap_data:
#    driver: local
```

Probably the fix is super easy, and probably there’s a totally obvious solution for anyone somewhat exeprienced with certificates besides from ”I used certbot for generating certificates for my website” … 😅

Disclaimer: I don’t really know what an issuer file is, nor how I would generate it; should maybe do some research on this.

**Comment by fooness** [comment]
I’m not even sure if `bitnami/openldap` even tries to use the specified certificate … is there any way to check if `slap.d`, if I’m not wrong, inside the container even knows about which certs/keys the service should be using? In other words, which `ldap.conf` or similar is being used inside the container?

**Comment by fooness** [comment]
After cleaning and pruning and removing basically *everything* docker related, I get the following error when starting the `docker-compose` stack for the first time; unfortunately I cannot figure out what’s wrong. When disabling TLS, everything starts up fine.

```
version: '2'

services:

  openldap--fix-permissions:
    #image: bitnami/openldap:latest
    image: docker.io/bitnami/openldap:2.6
    container_name: openldap--fix-permissions
    restart: "no"
    command: "chown -R 1001:1001 /bitnami"
    user: "root"
    volumes:
      - ./data/openldap:/bitnami

  openldap:
    image: docker.io/bitnami/openldap:2.6
    #image: bitnami/openldap:latest
    depends_on:
      - openldap--fix-permissions
    environment:
      - LDAP_ADMIN_USERNAME=admin
      - LDAP_ADMIN_PASSWORD=adminpassword
      - LDAP_USERS=user01,user02
      - LDAP_PASSWORDS=password1,password2
      - BITNAMI_DEBUG=true
      - LDAP_LOG_LEVEL=9
      - LDAP_PORT_NUMBER=1389
      - LDAP_ROOT=dc=example,dc=org
      - LDAP_USER_DC="users"
      - LDAP_GROUP="readers"
      - LDAP_ENABLE_TLS=yes
      - LDAP_LDAPS_PORT_NUMBER=1636
      - LDAP_TLS_CERT_FILE=/opt/bitnami/openldap/etc/certs/certificate.crt
      - LDAP_TLS_KEY_FILE=/opt/bitnami/openldap/etc/certs/privatekey.key
      - LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/lets-encrypt-r3.pem
      #- LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/isrg-root-x1-cross-signed.pem
      #- LDAP_TLS_CA_FILE=/opt/bitnami/openldap/etc/letsencrypt/isrgrootx1.pem
    ports:
      - "1389:1389"
      - "1636:1636"
    volumes:
      - ./data/openldap:/bitnami/openldap
      - ./data/letsencrypt:/opt/bitnami/openldap/etc/letsencrypt:ro
      - ./data/traefik-certs-dumper/openldap.example.org:/opt/bitnami/openldap/etc/certs:ro

#volumes:
#  openldap_data:
#    driver: local
```

```
docker-compose -f docker-compose.openldap.yml up
Starting openldap--fix-permissions ... done
Recreating openldap_1 ... done
Attaching to openldap--fix-permissions, openldap_1
openldap_1                   |  13:48:40.15 INFO  ==> ** Starting LDAP setup **
openldap--fix-permissions    |
openldap--fix-permissions exited with code 0
openldap_1                   |  13:48:40.18 INFO  ==> Validating settings in LDAP_* env vars
openldap_1                   |  13:48:40.19 INFO  ==> Initializing OpenLDAP...
openldap_1                   |  13:48:40.19 DEBUG ==> Ensuring expected directories/files exist...
openldap_1                   |  13:48:40.20 INFO  ==> Creating LDAP online configuration
openldap_1                   |  13:48:40.21 INFO  ==> Starting OpenLDAP server in background
openldap_1                   |  13:48:40.24 INFO  ==> Configure LDAP credentials for admin user
openldap_1                   | SASL/EXTERNAL authentication started
openldap_1                   | SASL username: gidNumber=0+uidNumber=1001,cn=peercred,cn=external,cn=auth
openldap_1                   | SASL SSF: 0
openldap_1                   | modifying entry "olcDatabase={2}mdb,cn=config"
openldap_1                   |
openldap_1                   | modifying entry "olcDatabase={2}mdb,cn=config"
openldap_1                   |
openldap_1                   | modifying entry "olcDatabase={2}mdb,cn=config"
openldap_1                   |
openldap_1                   | modifying entry "olcDatabase={1}monitor,cn=config"
openldap_1                   |
openldap_1                   |  13:48:40.25 INFO  ==> Configuring TLS
openldap_1                   | SASL/EXTERNAL authentication started
openldap_1                   | SASL username: gidNumber=0+uidNumber=1001,cn=peercred,cn=external,cn=auth
openldap_1                   | SASL SSF: 0
openldap_1                   | ldap_modify: Other (e.g., implementation specific) error (80)
openldap_1                   | modifying entry "cn=config"
openldap_1                   |
openldap_1 exited with code 80
```

**Comment by dgomezleon** [comment]
>Disclaimer: I don’t really know what an issuer file is, nor how I would generate it; should maybe do some research on this.

In my case, I created the certificates in another machine using [bnhelper-tool](https://docs.bitnami.com/general/how-to/generate-install-lets-encrypt-ssl/#use-the-bitnami-https-configuration-tool). This generated:

```
DOMAIN.crt     DOMAIN.issuer.crt    DOMAIN.json    DOMAIN.key
```

This [link](https://letsencrypt.org/certs/isrgrootx1.pem) also worked for me.

>Oh wait, I basically missed this step echo "127.0.0.1 DOMAIN" >> /etc/hosts in the container, instead I tried with localhost; when trying, though, I get: bash: /etc/hosts: Permission denied; I “entered” the container via: docker exec -it CONTAINER_ID /bin/bash

Yes, you need root privileges for that `docker exec -u root -it CONTAINER_ID bash`.

About the error `ldap_modify: Other (e.g., implementation specific) error (80)` , I have found a [post](https://askubuntu.com/questions/936382/openldap-error-configuring-starttls-ldap-modify-other-e-g-implementation-sp) that may help.

**Comment by fooness** [comment]
**tl;dr** I think we finally got it up and—almost—running with `ldaps://` via port `1636` working! 🎉

… and it was a permission problem all along—and still is when combined with `traefik-certs-dumper`; I suppose due to `bitnami/openldap` being conceptualized as non-root container, and therefore local files from our host operating system being, by default, not accessible from within `bitnami/openldap`.

I thought adding `user: root` to our `docker-compose.yml`, as outlined in the [bitnami docs](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers), would work-around this issue, but it does not; files/folders end up with id `1000` instead of `root`; and the problems persists.

Also: When there’s any problem with these permissions at the first start of `bitnami/openldap`, subsequent restarts—after fixing the permissions—don’t fix the problem, as the openldap configuration seems not to be updated/fixed, but keeps being broken. That means, after an unsuccessful first start of `bitnami/openldap`, we *always* had to delete the openldap data folder or volume … that alone made it quite complicated to debug this whole issue.

There’s some nasty timing issues going on as well; `traefik-certs-dumper` creates the necessary certificates as `root:root` … and it does not only once, but it keeps watching the `acme.json` managed by `traefik` and when it changes, the certificates are dumped again.

I tend to believe that it might be “easiest” to add another service which creates and maintains (i.e. extends/updates, if necessary) the certificate for *only* the openldap domain, and tell `traefik` *not* to create that certificate. I’m not sure, though, what’s the most simplistic way to add a `certbots`/`letsencrypt` service to the barebones `docker-compose.yml` you posted.

---

_Some comments to your previous message below, even though irrelevant for the subsequent discussion:_

> In my case, I created the certificates in another machine using [bnhelper-tool](https://docs.bitnami.com/general/how-to/generate-install-lets-encrypt-ssl/#use-the-bitnami-https-configuration-tool). This generated:
> 
> ```
> DOMAIN.crt     DOMAIN.issuer.crt    DOMAIN.json    DOMAIN.key
> ```

Thank you very much; I will have give this a read and will keep it in the back of my mind; seems useful!
 
> This [link](https://letsencrypt.org/certs/isrgrootx1.pem) also worked for me.

Very helpful you could confirm that using this certificate for `LDAP_TLS_CA_FILE` works in your case.

> > Oh wait, I basically missed this step echo "127.0.0.1 DOMAIN" >> /etc/hosts in the container, instead I tried with localhost; when trying, though, I get: bash: /etc/hosts: Permission denied; I “entered” the container via: docker exec -it CONTAINER_ID /bin/bash
> 
> Yes, you need root privileges for that `docker exec -u root -it CONTAINER_ID bash`.

Thank you! I searched quite some time for basically how to `sudo` in self-hosted bitnami containers without success.

> About the error `ldap_modify: Other (e.g., implementation specific) error (80)` , I have found a [post](https://askubuntu.com/questions/936382/openldap-error-configuring-starttls-ldap-modify-other-e-g-implementation-sp) that may help.

**Comment by dgomezleon** [comment]
Hi @fooness ,

Thanks for your feedback. I'm happy my previous comment was useful.

Checking your permissions issues, Wouldn't it be an option to use bitnami/openldap as a root container for you? [Link](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers)

**Comment by andirueckel** [comment]
Hello @dgomezleon,

> Checking your permissions issues, Wouldn't it be an option to use bitnami/openldap as a root container for you? [Link](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers)

Yeah, definitely, and I did try that; but unfortunately without any success; I shortly outlined this above:

>> I thought adding `user: root` to our `docker-compose.yml`, as outlined in the [bitnami docs](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers), would work-around this issue, but it does not; files/folders end up with id `1000` instead of `root`; and the problems persists.

If running the container (and everything withint) as legit `root` user, the problem would probably be fully worked-around.

**Comment by fooness** [comment]
Hello @dgomezleon,

> Checking your permissions issues, Wouldn't it be an option to use bitnami/openldap as a root container for you? [Link](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers)

Yeah, definitely, and I did try that; but unfortunately without any success; I shortly outlined this above:

>> I thought adding user: root to our docker-compose.yml, as outlined in the [bitnami docs](https://docs.bitnami.com/tutorials/work-with-non-root-containers#using-non-root-containers-as-root-containers), would work-around this issue, but it does not; files/folders end up with id 1000 instead of root; and the problems persists.

If running the container (and everything withint) as legit root user, the problem would probably be fully worked-around.
